go/adbc/driver/snowflake/driver.go

```diff
@@ -170,19 +171,30 @@ func quoteTblName(name string) string {
 	return "\"" + strings.ReplaceAll(name, "\"", "\"\"") + "\""
 }
 
+type DriverOption func(*gosnowflake.Config) error
```

Comment:
This definition is open enough that users can implement their own DriverOption funcs to change `*gosnowflake.Config` values arbitrarily. 
Whether that is undesired may be arguable, but if it is a concern, we can also create an unexported config struct that can both be extended and prevents from users defining custom functions:
```suggestion
type config struct {
	*gosnowflake.Config
}

type DriverOption func(*config) error
```

Reply:
Resolving this - see https://github.com/apache/arrow-adbc/pull/2558#discussion_r1972174338